src/ui/public/courier/fetch/__tests__/fetch_these.js

```diff
@@ -0,0 +1,84 @@
+describe('ui/courier/fetch/_fetch_these', () => {
+  const _ = require('lodash');
+  const sinon = require('auto-release-sinon');
+  const expect = require('expect.js');
+  const ngMock = require('ngMock');
+
+  let _Promise;
+  let $rootScope;
+  let fetchThese;
+  let request;
+  let requests;
+  let fakeResponses;
+
+  beforeEach(ngMock.module('kibana', (PrivateProvider) => {
+    function FakeResponsesProvider(Promise) {
+      fakeResponses = sinon.spy(function () {
+        return Promise.map(requests, mockRequest => {
+          return { mockRequest };
+        });
+      });
+      return fakeResponses;
+    }
+
+    PrivateProvider.swap(require('ui/courier/fetch/_call_client'), FakeResponsesProvider);
+    PrivateProvider.swap(require('ui/courier/fetch/_call_response_handlers'), FakeResponsesProvider);
+    PrivateProvider.swap(require('ui/courier/fetch/_continue_incomplete'), FakeResponsesProvider);
+  }));
+
+  beforeEach(ngMock.inject((Private, Promise, $injector) => {
+    _Promise = Promise;
+    $rootScope = $injector.get('$rootScope');
+    fetchThese = Private(require('ui/courier/fetch/_fetch_these'));
+    request = mockRequest();
+    requests = [ request ];
+  }));
+
+  context('when request has not started', () => {
+    beforeEach(() => _.forEach(requests, req => req.started = false));
```

Comment:
this should fail if requests is longer than one because `_.forEach()` stops when `false` is returned. Using `request.forEach()` prevents that behavior.